app/subscribers/zync_subscriber.rb

```diff
@@ -11,6 +11,6 @@ def initialize(job = ZyncWorker)
 
   # @param [ZyncEvent] event
   def after_commit(event)
-    job.perform_async(event.event_id, event.data)
+    job.perform_async(event.event_id, event.data) unless event.skip_background_sync?
```

Comment:
This is an ugly hack and perhaps we don't even need it. If we always go with the Sidekiq batch, maybe it's enough simply:

```ruby
class ZyncWorker
  def sync_dependencies(event_id)
    dependency_events = create_dependency_events(event_id)

    enqueue_dependency_batch(event_id, dependency_events) do
      publish_dependency_events(dependency_events)
    end
  end

def enqueue_dependency_batch(event_id, dependency_events, &block)
    batch = Sidekiq::Batch.new
    batch.description = "[ZyncWorker] Syncing dependencies first"
    batch.on(:complete, self.class, { 'event_id' => event_id })
    batch.jobs &block
  end
end
```

Reply:
Right, perhaps we could remove this `after_commit` completely.

![image](https://user-images.githubusercontent.com/11318903/57846164-e7118280-77d3-11e9-9ccb-701f69f54af7.png)

In here:
https://github.com/3scale/porta/blob/f70a1536c29aed56d9dd03fbfe470aca01bb4502/app/subscribers/publish_zync_event_subscriber.rb#L14-L25

We could change it for:
```ruby
def call(event)
  zync = case event
    # When the app is deleted the event caries all the needed information
    when Cinstances::CinstanceCancellationEvent then ZyncEvent.create(event)
    when ApplicationRelatedEvent then ZyncEvent.create(event, event.application)
    when OIDC::ProxyChangedEvent then ZyncEvent.create(event, event.proxy)
    when OIDC::ServiceChangedEvent then ZyncEvent.create(event, event.service)
    else raise "Unknown event type #{event.class}"
    end

  publisher.call(zync, 'zync')
  ZyncWorker.perform_async(zync.event_id, zync.data)
end
```